examples/native-fungible/src/contract.rs

```diff
@@ -0,0 +1,323 @@
+                owner: Some(owner),
+            };
+            system_api::transfer(None, account, amount);
+        }
+        Ok(ExecutionOutcome::default())
+    }
+
+    async fn execute_operation(
+        &mut self,
+        context: &OperationContext,
+        operation: Self::Operation,
+    ) -> Result<ExecutionOutcome<Self::Message>, Self::Error> {
+        match operation {
+            Operation::Transfer {
+                owner,
+                amount,
+                target_account,
+            } => {
+                Self::check_account_authentication(context.authenticated_signer, owner)?;
+                let account_owner = owner;
+                let owner = self.normalize_owner(owner);
+
+                let fungible_target_account = target_account;
+                let target_account = self.normalize_account(target_account);
+
+                system_api::transfer(Some(owner), target_account, amount);
+
+                Ok(self.get_transfer_outcome(account_owner, fungible_target_account, amount))
+            }
+
+            Operation::Claim {
+                source_account,
+                amount,
+                target_account,
+            } => {
+                Self::check_account_authentication(
+                    context.authenticated_signer,
+                    source_account.owner,
+                )?;
+
+                let fungible_source_account = source_account;
+                let fungible_target_account = target_account;
+
+                let source_account = self.normalize_account(source_account);
+                let target_account = self.normalize_account(target_account);
+
+                system_api::claim(source_account, target_account, amount);
+                Ok(
+                    self.get_claim_outcome(
+                        fungible_source_account,
+                        fungible_target_account,
+                        amount,
+                    ),
+                )
+            }
+        }
+    }
+
+    async fn execute_message(
+        &mut self,
+        context: &MessageContext,
+        message: Self::Message,
+    ) -> Result<ExecutionOutcome<Self::Message>, Self::Error> {
+        // Messages for now don't do anything, just pass messages around
+        match message {
+            Message::Credit {
+                amount: _,
+                target: _,
+                source: _,
+            } => {
+                // If we ever actually implement this, we need to remember
+                // to check if it's a bouncing message like in the fungible app
+                Ok(ExecutionOutcome::default())
+            }
+            Message::Withdraw {
+                owner,
+                amount,
+                target_account,
+            } => {
+                Self::check_account_authentication(context.authenticated_signer, owner)?;
+                Ok(self.get_transfer_outcome(owner, target_account, amount))
```

Comment:
I know the message doesn't do anything, but that `Credit` message we're creating here is really misleading if this message is bounced, isn't it?
Doesn't have to be in this PR, but I really think we shouldn't send it if the `Withdraw` was bounced.

Reply:
(As discussed, if they aren't tracked, they will never be bounced.)